### PR TITLE
feat(Notification Badge): Adds the notification badge component

### LINKED ISSUE #2021: Icon with badge (for notification drawer)
Introduce a component for an icon with badge. Specfically, this will be leveraged with the notification drawer. See: https://github.com/patternfly/patternfly-next/issues/1789

### PR DESCRIPTION
**Pull request by dlabaj**
**What**:
Adds the notification badge component to react core.  

**Additional issues**:
#2021

### CODE REVIEW

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/NotificationBadgeDemo/NotificationBadgeDemo.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+export class NotificationBadgeDemo extends React.Component <NotificationBadgeProps, NotificationBadgeDemoState>{
+  constructor(props) {
+    super(props);
+    this.state = {
+      isRead: false
+    };
+  }
+  onClick = () => {
+    this.setState({
+      isRead: true
```
Comment (redallen) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/NotificationBadgeDemo/NotificationBadgeDemo.tsx]: Could go with `isRead: !this.state.isRead` to allow toggling.

File: packages/patternfly-4/react-core/src/components/NotificationBadge/NotificationBadge.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+class SimpleNotificationBadge extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isRead: false
+    };
+    this.onClick = () => {
+      this.setState({
+        isRead: true
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/NotificationBadge/NotificationBadge.md]: Same with the example.

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/NotificationBadgeDemo/NotificationBadgeDemo.tsx (hunk 1 above)
Comment (dlabaj) [reply packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/NotificationBadgeDemo/NotificationBadgeDemo.tsx]: Thanks @redallen I figured once you set it you wouldn't want to show unread again, but if we want I can update the example to have it toggle. Just let me know I'm fine either way.

**Comment by dlabaj** [comment]
@mcarrano I don't have access to that visual spec above

**Comment by dlabaj** [comment]
@mcarrano Right now this is an external component that can also be used outside of the notification drawer.  Is the circle around it when it's toggled on suppose to be shown when it's not used with the notification drawer? 

@christiemolloy @mcoker I don't see the toggled on circle in the core example. https://pf4.patternfly.org/components/NotificationBadge/examples/ Is this being added in a modifier that we will get later?

**Comment by dlabaj** [comment]
@mcarrano Spoke with Mary and she's going to add that selected state when the drawer is open to this issue https://github.com/patternfly/patternfly-next/issues/1832.  When we get this in core and it is time to add it to react we will add that to the notification badge in the masthead at that time.  Does that sound good?

**Comment by christiemolloy** [comment]
This looks great @dlabaj and we hadn't addressed the circle background behind the icon yet because we still need to discuss how to best implement it in Core. I'm going to create a separate issue for it because patternfly/patternfly-next#1832 is large and addresses different concepts.

**Comment by mcarrano** [comment]
@dlabaj @christiemolloy this all makes sense and thanks for identifying the approach here. I that case I will go ahead and approve of this PR.

**Pull request merged**